### ISSUE TITLE
[BUG] `dark` and `light` in [data-theme] attribute of <html> tag were detected as className

### ISSUE THREAD

**Issue by hoangnhan2ka3** [soranoo/next-css-obfuscator#25]
**Type**
- [x] Incorrect Class Name Extraction
- [x] Incorrect JavaScript Obfuscation
- [ ] Incorrect HTML Obfuscation
- [ ] Incorrect CSS Obfuscation
- [ ] Others

**Checklist**
1. [x] Updated the package to the latest version
2. [x] Read all documentation
3. [x] No related issue
4. [x] Meaningful issue title

**Environment**
- Package Version: v.2.2.10
- Node Version: v.20.10.0

**Describe the bug**
`dark` and `light` in [data-theme] attribute of <html> tag were detected as className

**To Reproduce**
1. Use this package https://www.npmjs.com/package/next-app-theme
2. Check out the `./src/theme-script.tsx`

```tsx
type Theme = "dark" | "light";

declare global {
	interface Window {
		__theme: Theme;
		__onThemeChange: (theme: Theme) => void;
		__setPreferredTheme: (theme: Theme) => void;
	}
}

const code = function () {
	window.__onThemeChange = function () {};

	function setTheme(newTheme: Theme) {
		// document.documentElement.classList.remove(window.__theme);
		window.__theme = newTheme;
		preferredTheme = newTheme;
		document.documentElement.dataset.theme = newTheme;
		window.__onThemeChange(newTheme);
		// document.documentElement.classList.add(newTheme);
	}

	var preferredTheme;

	try {
		preferredTheme = localStorage.getItem("theme") as Theme;
	} catch (err) {}

	window.__setPreferredTheme = function (newTheme: Theme) {
		setTheme(newTheme);
		try {
			localStorage.setItem("theme", newTheme);
		} catch (err) {}
	};

	var darkQuery = window.matchMedia("(prefers-color-scheme: dark)");

	darkQuery.addEventListener("change", function (e) {
		window.__setPreferredTheme(e.matches ? "dark" : "light");
	});

	setTheme(preferredTheme || (darkQuery.matches ? "dark" : "light"));
};

const getTheme = `(${code})();`;

export function ThemeScript() {
	return <script dangerouslySetInnerHTML={{ __html: getTheme }} />;
}
```

**Expected behavior**
`[data-theme]` is not a `className` so... `dark` and `light` in `[data-theme]="dark"` or `[data-theme]="light"` is shouldn't be obfuscated.

![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/4f515e77-dfdb-419a-8f57-c1d264ece32e)

**Screenshots**

First load on the website maybe was right, but try add a `toggle` and press it:

![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/1fe7897f-ace8-4e4d-bf32-6c7283d7dfa6)

In `conversion.json`:

![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/feaf5bce-ac49-4038-ac58-ca3fc2f1a53f)

**Config**
```cjs
module.exports = {
  enable: true,
  mode: "random",
  buildFolderPath: ".next",
  classConversionJsonFolderPath: "./css-obfuscator",
  refreshClassConversionJson: false,

  classLength: 6,
  classPrefix: "n",
  classSuffix: "",

  classIgnore: ["no-transition", "opa-hidden", /nprogress*/, /__.*/, /os-*/],

  allowExtensions: [".jsx", ".tsx", ".js", ".ts", ".html", ".rsc"],

  contentIgnoreRegexes: [/\.jsxs\)\("\w+"/g],

  whiteListedFolderPaths: [],

  blackListedFolderPaths: ["./.next/cache", "./src", "./public", "./tests"],

  enableMarkers: false,
  markers: ["obf"],
  removeMarkersAfterObfuscated: true,

  removeOriginalCss: true,

  generatorSeed: "84817818898",

  //! Experimental feature (Alpha)
  enableJsAst: false,

  logLevel: "info",
};
```

**Related files**
`Obfuscated` version:
[97fb199871848870.css.txt](https://github.com/soranoo/next-css-obfuscator/files/14322690/97fb199871848870.css.txt)

**Demos (if any)**
N/A

**Additional context**
...

**Comment by soranoo** [comment]
As a temporary solution, add the following to the config
```js
classIgnore = ["dark", "light"]
```

I will get back to the issue after a month.

**Comment by hoangnhan2ka3** [comment]
Yes, I will. My issues all want to aim for a `config file` that doesn't require much adjustment and is as simple as possible, I hope you understand.

Also, what are you busy with in the next month?

**Comment by soranoo** [comment]
Mate, I can't eat with open source.

**Comment by hoangnhan2ka3** [comment]
Sooo embarrassing, very sorry, I'm just a student, I can't help you any more than helping your package become more perfect so that more people can pay attention to it, at some point it will definitely work, keep trying. Come on up.

This is the first time in half a year that I have come into contact with Front-end development and I am so interested in a package.

I still dream that one day Facebook will buy this package, they created StyleX because they somehow hate Tailwind's cumbersome classes :)))

**Comment by hoangnhan2ka3** [comment]
Something wrong :)))

![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/848642ea-eb18-4a2b-ab66-753f7775056b)

![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/cc30a4c6-275c-47c1-b3df-df73dc484b60)

If you need a real demo, I will update to the previous repo I shared with you after I finished lunch

**Comment by soranoo** [comment]
May I have the related HTML file(`.html`) (both before and after obfuscation) and the `conversion.json`?

**Comment by hoangnhan2ka3** [comment]
All expected behavior is at [hoangnhan.uk](https://hoangnhan.uk)

**Comment by soranoo** [comment]
And what is your expected output? for the html file

**Comment by hoangnhan2ka3** [comment]
No, maybe you misunderstood something.

The `data-theme` attribute only appears at runtime, but in the `static html` file it will not.

`data-theme` appears through this script in the `head` tag (which was created by `theme-script.ts`):

![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/4c907f14-a0fb-4382-8995-8f875f79be61)

**Comment by hoangnhan2ka3** [comment]
Basically after `run build`, everything is fine, however when running with the current `2.2.10` version, as I said above

> **Expected behavior**
> `[data-theme]` is not a `className` so... `dark` and `light` in `[data-theme]="dark"` or `[data-theme]="light"` is shouldn't be obfuscated.
> 
> ![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/4f515e77-dfdb-419a-8f57-c1d264ece32e)
> 
> **Screenshots**
> 
> First load on the website maybe was right, but try add a `toggle` and press it:
> 
> ![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/1fe7897f-ace8-4e4d-bf32-6c7283d7dfa6)
> 
> In `conversion.json`:
> 
> ![image](https://github.com/soranoo/next-css-obfuscator/assets/147973044/feaf5bce-ac49-4038-ac58-ca3fc2f1a53f)

And things are even worse in the `canary` version...

**Comment by hoangnhan2ka3** [comment]
Solve without https://github.com/soranoo/next-css-obfuscator/issues/25, but problem remain https://github.com/soranoo/next-css-obfuscator/issues/30

**Comment by soranoo** [comment]
> Because `2.2.11` and `2.2.11-beta.2` broke the site in #27 so I'm not able to check this issue yet
> 
> ![image](https://private-user-images.githubusercontent.com/147973044/311112728-76021b14-2364-41e0-94ab-749f71bafe81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxOTYzODEsIm5iZiI6MTcxMDE5NjA4MSwicGF0aCI6Ii8xNDc5NzMwNDQvMzExMTEyNzI4LTc2MDIxYjE0LTIzNjQtNDFlMC05NGFiLTc0OWY3MWJhZmU4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMVQyMjI4MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYTBkMGJlM2ZiMWE1NjAwZTVlYmQ2NDU1NjQwZWZmOWY0ZGRiNjQ0NDM0YTg3ODBlOTVjOGE5NWEyOTU2YmIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LNZIxthbAGX1nXVSLvDjNZ4yrNZY2gGuA1t0tmFOujk)

[For future reference]

The error - "Minified react error #148" is due to 
1. Missing HTML comment (e.t. `<!--$-->`)
2. Invalid HTML tag (eg. `<br/>`valid, `<br></br>`invalid)

**Comment by hoangnhan2ka3** [comment]
No, I don't have that. The proof is that when you revert the update, it still runs normally and without any errors.

**Comment by soranoo** [comment]
Im still working on it.

**Comment by hoangnhan2ka3** [comment]
Solved in `v2.2.14-beta.4`
